oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/Cursors.java

```diff
@@ -339,7 +340,9 @@ private void fetchNext() {
                     readCount++;
                     if (readCount % 1000 == 0) {
                         FilterIterators.checkReadLimit(readCount, settings);
-                        LOG.warn("Traversed " + readCount + " nodes with filter " + filter + "; consider creating an index or changing the query");
+                        String caller = IndexUtils.getCaller(this.settings.getIgnoredClassNamesInCallTrace());
+                        LOG.warn("Traversed {} nodes with filter {} called by {}; consider creating an index or changing the query" , 
+                                new Object[] {readCount, filter, caller});
```

Comment:
the array creation could be avoided since the method receives a `varargs`
```suggestion
                        LOG.warn("Traversed {} nodes with filter {} called by {}; consider creating an index or changing the query" ,
                                readCount, filter, caller);
```